apps/web/pages/api/basenames/[name]/assets/coverImage.png.tsx

```diff
@@ -19,6 +23,25 @@ export default async function handler(request: NextRequest) {
   const username = url.searchParams.get('name') ?? 'yourname';
   const domainName = isDevelopment ? `${url.protocol}//${url.host}` : 'https://www.base.org';
   const profilePicture = getUserNamePicture(username);
+  const chainId = url.searchParams.get('chainId') ?? base.id;
+  let imageSource = domainName + profilePicture.src;
+
+  // NOTE: Do we want to fail if the name doesn't exists?
+  try {
+    const nameHash = namehash(username);
+    const client = getBasenamePublicClient(Number(chainId));
+    const avatar = await client.readContract({
+      abi: L2ResolverAbi,
+      address: USERNAME_L2_RESOLVER_ADDRESSES[Number(chainId)],
+      args: [nameHash, UsernameTextRecordKeys.Avatar],
+      functionName: 'text',
+    });
```

Comment:
I seem to remember there being a way to get better perf from this by using `call` directly and only modifying the data you're calling with so you don't need to process the abi for each invocation

Reply:
can't find the reference for this!